src/tool/subcommands/api_cmd.rs

```diff
@@ -764,6 +767,31 @@ fn snapshot_tests(store: &ManyCar, n_tipsets: usize) -> anyhow::Result<Vec<RpcTe
                 )));
             }
         }
+
+        // Get deals (shuffled)
+        let deals = {
+            let state = StateTree::new_from_root(store.clone(), tipset.parent_state())?;
+            let actor = state
+                .get_actor(&Address::MARKET_ACTOR)?
+                .context("Market actor not found")?;
+            let market_state = market::State::load(&store, actor.code, actor.state)?;
+            let proposals = market_state.proposals(&store)?;
+            let mut deals = vec![];
+            proposals.for_each(|deal_id, _| {
+                deals.push(deal_id);
+                Ok(())
+            })?;
+            deals.shuffle(&mut OsRng);
```

Comment:
I'm not sure about this. Tests need to be deterministic if possible. Otherwise, we may end up with flaky tests that are not reproducible.

If we need shuffling, using a pre-determined seed for the RNG might be a way out. What do you think?

Reply:
But given the same snapshot (e.g., in a local test), making the seed constant would make the test reproducible, right? Or would it still be completely random from network conditions?

Reply:
> But given the same snapshot (e.g., in a local test), making the seed constant would make the test reproducible, right?

@LesnyRumcajs That's true, I can remove the shuffle to make it reproducible for a fixed snapshot.